cli/src/pcluster/schemas/common_schema.py

```diff
@@ -65,6 +65,18 @@ def validate_no_reserved_tag(tags):
                 raise ValidationError(message=f"The tag key prefix '{PCLUSTER_PREFIX}' is reserved and cannot be used.")
 
 
+def validate_no_duplicate_tag(tags):
+    """Validate there is no duplicate tag keys in the same tag section."""
+    all_tags = set()
+    for tag in tags:
+        tag_key = tag.key
+        if tag_key in all_tags:
+            raise ValidationError(
+                f"Duplicate tag key ({tag_key}) detected. Tags should not have same tag key with the Tags section."
+            )
+        all_tags.add(tag_key)
+
```

Comment:
For detecting duplicated keys, it is checked in the same way as we check duplicated instance types, in the case we find duplicate tag, we can raise the error instead of going through all the tag keys, this can reduce the validation time a bit when there's duplicate keys.
The text message is updated